tests/CMakeLists.txt

```diff
@@ -46,17 +45,16 @@ set(ENSMALLEN_TESTS_SOURCES
 )
 
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
-add_executable(${PROJECT_NAME} ${ENSMALLEN_TESTS_SOURCES})
-
-target_link_libraries(${PROJECT_NAME} ${ARMADILLO_LIBRARIES})
+add_executable(ensmallen_tests ${ENSMALLEN_TESTS_SOURCES})
+target_link_libraries(ensmallen_tests PRIVATE ensmallen)
 
 # Copy test data into place.
-add_custom_command(TARGET ${PROJECT_NAME}
+add_custom_command(TARGET ensmallen_tests
   POST_BUILD
   COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR}/data/
       ${CMAKE_BINARY_DIR}/data/
 )
 
 enable_testing()
 add_test(NAME ${PROJECT_NAME} COMMAND ${PROJECT_NAME}
```

Comment:
I think the unset project name may be why it's not finding the executable it wants to. You could also probably fix this by changing PROJECT_NAME to the literal "ensmallen_tests" if you don't want to set a project name here.

Reply:
Ah I thought I had committed this, thanks for pointing it out :)